pkg/virtctl/create/vm/vm_test.go

```diff
@@ -939,6 +942,183 @@ chpasswd: { expire: False }`
+			Entry("explicit type and default method, user from flag, nocloud default", "type:ssh,src:my-keys", verifyNoCloudGAManageSSHAndUser, setFlag(UserFlag, "myuser")),
+			Entry("explicit type and default method, user from flag, nocloud explicit", "type:ssh,src:my-keys", verifyNoCloudGAManageSSHAndUser, setFlag(UserFlag, "myuser"), setFlag(CloudInitFlag, CloudInitNoCloud)),
+			Entry("explicit type and default method, user from flag, configdrive explicit", "type:ssh,src:my-keys", verifyConfigDriveGAManageSSHAndUser, setFlag(UserFlag, "myuser"), setFlag(CloudInitFlag, CloudInitConfigDrive)),
+			Entry("explicit type and default method, user from flag, cloud-init none", "type:ssh,src:my-keys", verifyCloudInitNone, setFlag(UserFlag, "myuser"), setFlag(CloudInitFlag, CloudInitNone)),
+			Entry("explicit type and default method, user from flag, GAManageSSH false", "type:ssh,src:my-keys", verifyNoCloudUser, setFlag(UserFlag, "myuser"), setFlag(GAManageSSHFlag, "false")),
+			Entry("explicit type and method, user from param, nocloud default", "type:ssh,src:my-keys,method:ga,user:myuser", verifyNoCloudGAManageSSH),
+			Entry("explicit type and method, user from param, nocloud explicit", "type:ssh,src:my-keys,method:ga,user:myuser", verifyNoCloudGAManageSSH, setFlag(CloudInitFlag, CloudInitNoCloud)),
+			Entry("explicit type and method, user from param, configdrive explicit", "type:ssh,src:my-keys,method:ga,user:myuser", verifyConfigDriveGAManageSSH, setFlag(CloudInitFlag, CloudInitConfigDrive)),
+			Entry("explicit type and method, user from param, cloud-init none", "type:ssh,src:my-keys,method:ga,user:myuser", verifyCloudInitNone, setFlag(CloudInitFlag, CloudInitNone)),
+			Entry("explicit type and method, user from param, GAManageSSH false", "type:ssh,src:my-keys,method:ga,user:myuser", verifyCloudInitNone, setFlag(GAManageSSHFlag, "false")),
+			Entry("explicit type and method, user from flag, nocloud default", "type:ssh,src:my-keys,method:ga", verifyNoCloudGAManageSSHAndUser, setFlag(UserFlag, "myuser")),
+			Entry("explicit type and method, user from flag, nocloud explicit", "type:ssh,src:my-keys,method:ga", verifyNoCloudGAManageSSHAndUser, setFlag(UserFlag, "myuser"), setFlag(CloudInitFlag, CloudInitNoCloud)),
+			Entry("explicit type and method, user from flag, configdrive explicit", "type:ssh,src:my-keys,method:ga", verifyConfigDriveGAManageSSHAndUser, setFlag(UserFlag, "myuser"), setFlag(CloudInitFlag, CloudInitConfigDrive)),
+			Entry("explicit type and method, user from flag, cloud-init none", "type:ssh,src:my-keys,method:ga", verifyCloudInitNone, setFlag(UserFlag, "myuser"), setFlag(CloudInitFlag, CloudInitNone)),
+			Entry("explicit type and method, user from flag, GAManageSSH false", "type:ssh,src:my-keys,method:ga", verifyNoCloudUser, setFlag(UserFlag, "myuser"), setFlag(GAManageSSHFlag, "false")),
```

Comment:
supersupernit - I'm really not a fan of massive tables like this with so many test cases being handled. You don't need to change anything more of a personal issue with our approach to using `Tables` sometimes, breaking this into a `Context` with multiple more targetted `Tables` would be better IMHO.

Reply:
What about https://onsi.github.io/ginkgo/#generating-subtree-tables ?